### PR TITLE
Implement the app as specified in #7

### PR DESCRIPTION
**Pull request by ianb**
I added on all the server routines, so I think it's pretty much complete now. Not the fanciest UI, but workable.

### CODE REVIEW

File: server/static/page.jsx [hunk 1]
```diff
@@ -1,8 +1,404 @@
...
+    model.lookingUpShot.shotSearching = true;
+    render();
+    let resp = await fetch(`/lookup-shot?term=${encodeURIComponent(searchTerm)}`);
+    model.lookingUpShot.shotSearching = false;
+    if (resp.status === 404) {
+      model.lookingUpShot.notFound = true;
+      render();
+      return;
+    }
+    if (! resp.ok) {
```
Comment (6a68) [inline server/static/page.jsx]: Not sure it really matters, but should `model.lookingUpShot.notFound` be `true` in this case, too?

File: server/static/page.jsx [hunk 2]
```diff
@@ -1,8 +1,404 @@
...
+    model.lookingUpDevice.deviceSearching = true;
+    render();
+    let resp = await fetch(`/lookup-device?deviceId=${encodeURIComponent(deviceId)}`);
+    model.lookingUpDevice.deviceSearching = false;
+    if (resp.status === 404) {
+      model.lookingUpDevice.notFound = true;
+      render();
+      return;
+    }
+    if (! resp.ok) {
```
Comment (6a68) [inline server/static/page.jsx]: Same question here, I guess, about the `notFound` state

File: server/static/page.jsx [hunk 3]
```diff
@@ -1,8 +1,404 @@
...
+        id="block-shot-input"
+        type="text" />
+        <select
+        ref={select => this.select = select} required>
+          <option value="">Select block reason</option>
+          <option value="none">Clear block</option>
+          <option value="dmca">DMCA</option>
+          <option value="usererror">User error</option>
+          <option value="illegal">Illegal</option>
+          <option value="illegal-quarantine">Illegal + quarantine</option>
```
Comment (6a68) [inline server/static/page.jsx]: Same question here about possible states including / not including 'watchdog'. Should we pull this list from the schema? If not, maybe file a bug on screenshots to comment in the schema about updating the admin interface when the enum changes...

**Comment by 6a68** [comment]
@ianb Oh, one other question - Do we need to include information about the user who is clicking in the form? IP address, other stuff ,maybe?

**Comment by ianb** [comment]
#11 (which this doesn't close) covers logging/auditing. We'll have to add that, yes.

File: server/server.js [hunk 4]
```diff
@@ -19,6 +20,163 @@ app.use(function(error, req, res, next) {
...
+}));
+
+app.post("/block-shot", wrap(async (req, res) => {
+  let shotId = await shotIdFromTerm(req.body.term);
+  let reason = req.body.reason; // Could be none, dmca, usererror, illegal, illegal-quaratine
+  if (!shotId) {
+    res.status(404).send("Not found");
+    return;
+  }
+  if (!["none", "dmca", "usererror"].includes(reason)) {
```
[the remark this answers is not in the document]
Comment (ianb) [reply server/server.js]: I'm working with the code as it is, based on past procedure (e.g., as described [here](https://github.com/mozilla-services/screenshots/blob/master/docs/takedown.md)). Once https://github.com/mozilla-services/screenshots/issues/4347 is done then this should be revisited.

**Pull request merged**